### PR TITLE
Bug 1328659 - Prepare MySQL database for emoji by performing row format conversion

### PR DESCRIPTION
**Pull request by dylanwh**

### CODE REVIEW

File: scripts/prepare-db-for-emoji.pl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+BEGIN { Bugzilla->extensions }
+
+# set Bugzilla usage mode to USAGE_MODE_CMDLINE
+Bugzilla->usage_mode(USAGE_MODE_CMDLINE);
+
+my $dbh = Bugzilla->dbh;
+
+my %mysql_var = map { @$_ } @{ $dbh->selectall_arrayref(q(SHOW GLOBAL VARIABLES LIKE 'innodb_%')) };
+my %wanted = (
+    innodb_file_format    => 'Barracuda',
```
Comment (floatingatoll) [inline scripts/prepare-db-for-emoji.pl]: Why specifically Barracuda?

File: scripts/prepare-db-for-emoji.pl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        $can_convert = 0;
+    }
+}
+die "Unable to convert from COMPACT to DYNAMIC\n" unless $can_convert;
+
+my $tables = $dbh->selectall_arrayref('SHOW TABLE STATUS');
+
+foreach my $table (@$tables) {
+    my ($table, undef, undef, $row_format) = @$table;
+    if ($row_format ne 'Dynamic') {
```
Comment (floatingatoll) [inline scripts/prepare-db-for-emoji.pl]: Why is this one Dynamic and all the rest uppercase?

File: scripts/prepare-db-for-emoji.pl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+);
+
+my $can_convert = 1;
+foreach my $key (keys %wanted) {
+    unless ($mysql_var{$key} eq $wanted{$key}) {
+        warn "$key must be set to $wanted{$key} (current value is $mysql_var{$key})\n";
+        $can_convert = 0;
+    }
+}
+die "Unable to convert from COMPACT to DYNAMIC\n" unless $can_convert;
```
Comment (floatingatoll) [inline scripts/prepare-db-for-emoji.pl]: “Until the above keys return the expected values”?

File: scripts/prepare-db-for-emoji.pl (hunk 1 above)
Comment (dylanwh) [reply scripts/prepare-db-for-emoji.pl]: Only Barracuda supports Dynamic row format.
Antelope is the default in 5.5/5.6 and only supports Compact.

**Pull request closed without merging**